framework/src/base/MooseApp.C

```diff
@@ -797,6 +796,34 @@ MooseApp::setupOptions()
       _action_warehouse.setFinalTask("split_mesh");
     }
     _action_warehouse.build();
+
+    _file_base_set_by_user = false;
```

Comment:
This Boolean should be set in the constructor, otherwise reading this value in the accessor you added could result in a Valgrind error if it's accessed before this method is called.

Reply:
Actually it's worse than that. It's quite possible to fall through this routine without ever setting this Boolean creating a real Valgrind issue.